python/social_bot/envs/embodied_teacher.py

```diff
@@ -120,11 +120,17 @@ def __init__(self,
         # insert teacher model
         self.insert_model(
             model=agent_type, name="teacher", pose=initial_teacher_pose)
+
         # set up teacher joints
+        self._teacher_cfg = self._agent_cfg
         self._teacher_joints = []
-        for joint in self._agent_joints:
+        print(self._agent_joints)
+        for joint in self._teacher_cfg['control_joints']:
             self._teacher_joints.append("teacher::" + joint)
         self._teacher_embodied = self._world.get_agent('teacher')
+        # TODO: this report error "unable to find joint"
+        self._teacher_control_range = self._set_joints(
+            self._teacher_embodied, self._teacher_joints, self._teacher_cfg)
```

Comment:
Feel we can put cfg, joints, agent, control_range into one single class.

Reply:
I was also thinking about this. 
A name of Agent for the new class is fine. Current agent can be renamed to an internal self._actor of Agent.

Reply:
Let's do this in another PR.